src/controllers/controllermanager.cpp

```diff
@@ -392,7 +392,17 @@ void ControllerManager::slotSavePresets(bool onlyActive) {
         if (onlyActive && !pController->isOpen()) {
             continue;
         }
+        ControllerPresetPointer pPreset = pController->getPreset();
+        DEBUG_ASSERT(!pPreset);
+
         QString deviceName = sanitizeString(pController->getName());
+        if (!pPreset->isDirty()) {
+            qWarning()
+                    << "Preset for device" << deviceName
+                    << "is not dirty, no need to save it to the user presets.";
+            continue;
+        }
```

Comment:
Could you append " (edited)" to the name of the preset here?

Reply:
Maybe this isn't the place to do it. The code that handles the mapping editing tables should change the name so that " (edited)" doesn't get appended repeatedly when the user edits multiple times.

Reply:
Check if it's a user or system mapping. If it's a user mapping, don't append " (edited)"

Reply:
Good idea. That way we don't end up with "My Mapping (edited) (edited) (edited)".